### PR TITLE
Updation of delfin documentation 

### PR DESCRIPTION
**Pull request by vineela1999**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind new feature
> /kind bug fix
> /kind cleanup
> /kind revert change
> /kind design

kind documentation
> /kind enhancement

**What this PR does / why we need it**:
This PR has the documentation updates of delfin 
**Which issue(s) this PR fixes**:

Fixes #

**Test Report Added?**:
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:

kind TESTED
> /kind NOT-TESTED

**Test Report**:

**Special notes for your reviewer**:

### CODE REVIEW

File: content/guides/developer-guides/delfin/driver-developer-guide/_index.md [hunk 1]
```diff
@@ -292,17 +300,162 @@ Existing Delfin Drivers for reference:
     def collect_perf_metrics(self, context, storage_id, metrics, start, stop):
         pass
 
-    def get_capabilities(self, context):
+    def get_capabilities(context, filters=None):
+        pass
+	
+    def list_storage_host_initiators(self, context):
+        storage_host_initiators_list = []
+        for idx in range(1, 3):
```
Comment (joseph-v) [inline content/guides/developer-guides/delfin/driver-developer-guide/_index.md:308]: Please remove for loop and make it simple list

File: content/guides/developer-guides/delfin/driver-developer-guide/_index.md [hunk 2]
```diff
@@ -292,17 +300,162 @@ Existing Delfin Drivers for reference:
...
+                "wwn": "wwn_" + str(idx),
+                "status": "Normal",
+                "native_storage_host_id": "storage_host_" + str(idx),
+            }
+            storage_host_initiators_list.append(s)
+        return storage_host_initiators_list
+
+    def list_storage_hosts(self, context):
+        storage_hosts_list = []
+        for idx in range(1, 3):
```
Comment (joseph-v) [inline content/guides/developer-guides/delfin/driver-developer-guide/_index.md:324]: Same as above

File: content/guides/developer-guides/delfin/driver-developer-guide/_index.md [hunk 3]
```diff
@@ -292,17 +300,162 @@ Existing Delfin Drivers for reference:
...
+            "native_storage_host_group_id": "storage_host_group_1",
+            "storage_hosts": "host1,host2"
+        },
+            {
+                "name": "storage_host_group_2",
+                "description": "storage_host_group_2",
+                "storage_id": self.storage_id,
+                "native_storage_host_group_id": "storage_host_group_2",
+                "storage_hosts": "host3,host4"
+            }]
```
Comment (joseph-v) [inline content/guides/developer-guides/delfin/driver-developer-guide/_index.md:351]: Use following formatting for better readability

```suggestion
        host_groups = [
            {
                "name": "storage_host_group_1",
                "description": "storage_host_group_1",
                "storage_id": self.storage_id,
                "native_storage_host_group_id": "storage_host_group_1",
                "storage_hosts": "host1,host2"
            },
            {
                "name": "storage_host_group_2",
                "description": "storage_host_group_2",
                "storage_id": self.storage_id,
                "native_storage_host_group_id": "storage_host_group_2",
                "storage_hosts": "host3,host4"
            }
        ]
```

File: content/guides/developer-guides/delfin/driver-developer-guide/_index.md [hunk 4]
```diff
@@ -292,17 +300,162 @@ Existing Delfin Drivers for reference:
...
+
+        storage_host_group_relation_list = [{
```
Comment (joseph-v) [inline content/guides/developer-guides/delfin/driver-developer-guide/_index.md:353]: Use formatting as above

File: content/guides/developer-guides/delfin/driver-developer-guide/_index.md [hunk 5]
```diff
@@ -292,17 +300,162 @@ Existing Delfin Drivers for reference:
...
+
+    def list_port_groups(self, context):
+        port_group_list = [{
+            "name": "storage_port_group_1",
+            "description": "storage_port_group_1",
+            "storage_id": self.storage_id,
+            "native_port_group_id": "storage_port_group_1",
+            "ports": "port_1,port_2"
+        }]
+        port_group_relation = [{
```
Comment (joseph-v) [inline content/guides/developer-guides/delfin/driver-developer-guide/_index.md:377]: Update formatting

File: content/guides/developer-guides/delfin/driver-developer-guide/_index.md [hunk 6]
```diff
@@ -292,17 +300,162 @@ Existing Delfin Drivers for reference:
...
+                'native_volume_group_id': "storage_volume_group_1",
+                'native_volume_id': "volume1"
+
+            },
+            {
+                'storage_id': self.storage_id,
+                'native_volume_group_id': "storage_volume_group_1",
+                'native_volume_id': "volume2"
+
+            }]
```
Comment (joseph-v) [inline content/guides/developer-guides/delfin/driver-developer-guide/_index.md:414]: Update formatting

File: content/guides/developer-guides/delfin/driver-developer-guide/_index.md [hunk 7]
```diff
@@ -292,17 +300,162 @@ Existing Delfin Drivers for reference:
...
+        result = {
+            'volume_groups': volume_group_list,
+            'vol_grp_vol_rels': volume_group_relation
+        }
+        return result
+
+    def list_masking_views(self, context):
+        masking_view_list = [{
```
Comment (joseph-v) [inline content/guides/developer-guides/delfin/driver-developer-guide/_index.md:422]: update formatting

File: content/guides/developer-guides/delfin/driver-developer-guide/_index.md (hunk 2 above)
Comment (vineela1999) [reply content/guides/developer-guides/delfin/driver-developer-guide/_index.md]: changed

File: content/guides/developer-guides/delfin/driver-developer-guide/_index.md (hunk 3 above)
Comment (vineela1999) [reply content/guides/developer-guides/delfin/driver-developer-guide/_index.md]: updated

File: content/guides/developer-guides/delfin/driver-developer-guide/_index.md (hunk 4 above)
Comment (vineela1999) [reply content/guides/developer-guides/delfin/driver-developer-guide/_index.md]: updated

File: content/guides/developer-guides/delfin/driver-developer-guide/_index.md (hunk 5 above)
Comment (vineela1999) [reply content/guides/developer-guides/delfin/driver-developer-guide/_index.md]: updated

File: content/guides/developer-guides/delfin/driver-developer-guide/_index.md (hunk 6 above)
Comment (vineela1999) [reply content/guides/developer-guides/delfin/driver-developer-guide/_index.md]: updated

File: content/guides/developer-guides/delfin/driver-developer-guide/_index.md [hunk 8]
```diff
@@ -292,17 +300,200 @@ Existing Delfin Drivers for reference:
...
+        return result
+
+    def list_port_groups(self, context):
+        port_group_list = [
+            {
+                "name": "port_group_1",
+                "description": "port_group_1",
+                "storage_id": self.storage_id,
+                "native_port_group_id": "storage_port_group_1",
+                "ports": "port_1"
```
Comment (joseph-v) [inline content/guides/developer-guides/delfin/driver-developer-guide/_index.md:407]: Do we need port_2 here?

Comment (vineela1999) [reply content/guides/developer-guides/delfin/driver-developer-guide/_index.md]: yes have added

**Review by wisererik — APPROVED** [review]

**Review by joseph-v — APPROVED** [review]

**Pull request merged**